src/main/kotlin/com/ex_dock/ex_dock/database/service/DatabaseBackupVerticle.kt

```diff
@@ -0,0 +1,254 @@
+    val parts = time.split(":")
+    return (parts[0].toInt() * 60 * 60 + parts[1].toInt() * 60) * 1000L
+  }
+
+  /**
+   * Removes the oldest backup if the number of backups is greater than the maximum number of backups allowed
+   */
+  private fun organizeBackups() {
+    val backupDirectory = System.getProperty("user.dir") + "\\database\\backup"
+    val folder = File(backupDirectory)
+    val listOfFiles = folder.listFiles()
+    val fileDateList: MutableList<String> = emptyList<String>().toMutableList()
+    var oldestDate = ""
+
+    // Skip organizing when the number of backups is lower than the maximum number of backups allowed
+    if (listOfFiles!!.size >= savedBackups) {
+      for (f in listOfFiles) {
+        val fileDateString = f.name
+          .replace("backup_", "")
+          .replace(".sql", "")
+        fileDateList.add(fileDateString)
+      }
+
+      for (dateString in fileDateList) {
+        oldestDate = if (oldestDate != "") {
+          val dateParts = dateString.split("_")
+          val oldestDateParts = oldestDate.split("_")
+
+          checkIfOlderFile(
+                dateParts,
+                oldestDateParts,
+                dateString,
+                oldestDate)
+        } else {
+          dateString
+        }
+      }
+
+      val oldestBackup = File("$backupDirectory\\backup_$oldestDate.sql")
+
+      if (oldestBackup.delete()) {
+        println("Deleted oldest backup from: $oldestDate")
+      } else {
+        println("Could not delete oldest backup from: $oldestDate")
+      }
+    }
+  }
+
+  /**
+   * Checks the backup name and returns the oldest backup date
+   */
+  private fun checkIfOlderFile(dateParts: List<String>,
+                               oldestDateParts: List<String>,
+                               dateString: String,
+                               oldestDate: String): String {
+    // Check if the Year is older
+    if (dateParts[2].toInt() < oldestDateParts[2].toInt()) {
+      return dateString
+
+      // Check if the Year is the same
+    } else if (dateParts[2].toInt() == oldestDateParts[2].toInt()) {
+
+      // Check if the Month is older
+      if (dateParts[1].toInt() < oldestDateParts[1].toInt()) {
+        return dateString
+
+        // Check if the Month is the same
+      } else if (dateParts[1].toInt() == oldestDateParts[1].toInt()) {
+
+        // Check if the Day is older
+        if (dateParts[0].toInt() < oldestDateParts[0].toInt()) {
+          return dateString
+
+          // Check if the Day is the same
+        } else if (dateParts[0].toInt() == oldestDateParts[0].toInt()) {
+
+          // Check if the Hour is older
+          if (dateParts[3].toInt() < oldestDateParts[3].toInt()) {
+            return dateString
+
+            // Check if the Hour is the same
+          } else if (dateParts[3].toInt() == oldestDateParts[3].toInt()) {
+            return dateString
+          }
+        }
+      }
+    }
+
+    return oldestDate
```

Comment:
Since this if-else structure uses returns, all the else's are unnecessary. Please remove them for clarity.